Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,501 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// A protocol for message content delegate responsible for action handling.
+///
+/// When custom message content view is created, the protocol that inherits from this one
+/// should be created if an action can be taken on the new content view.
+public protocol MessageContentViewDelegate: AnyObject {
+    func didTapOnErrorIndicator(at indexPath: IndexPath)
+    func didTapOnThread(at indexPath: IndexPath)
+    func didTapOnQuotedMessage(at indexPath: IndexPath)
+}
+
+/// A view that displays the message content.
+public typealias MessageContentView = _MessageContentView<NoExtraData>
+
+/// A view that displays the message content.
+open class _MessageContentView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+    /// The current layout options of the view.
+    /// When this value is set the subviews are instantiated and laid out just once based on
+    /// the received options.
+    private var layoutOptions: MessageLayoutOptions!
+
+    // MARK: Content && Actions
+
+    /// The index path of the cell which displays the current content view.
+    public var indexPath: IndexPath?
```

Comment:
If we decide to fix it one time this would be a breaking change. Do you think it worth trying to move the `indexPath` to the cell OR capture somehow when we set the delegate? 🤔

Reply:
Updated delegate method names and added docs there. 
I still think that it can/should be done without an optional `indexPath` field in the content view but it's gonna be easier to try different implementations when this content view is used in the message list and we will be working on action forwarding.